.github/workflows/test-template.yml

```diff
@@ -0,0 +1,101 @@
+name: Test Template
+
+on:
+  workflow_call:
+    inputs:
+      tags:
+        required: true
+        type: string
+      coverprofile:
+        required: true
+        type: string
+      binary_name:
+        required: true
+        type: string
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    container:
+      image: hub.opensciencegrid.org/pelican_platform/pelican-dev:latest-itb
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 20
+    - name: Fetch tags
+      run: |
+        git config --global --add safe.directory /__w/pelican/pelican
+        git fetch --force --tags
+    - name: Cache Next.js
+      uses: actions/cache@v4
+      with:
+        path: |
+          ~/.npm
+          ${{ github.workspace }}/.next/cache
+        key: ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('**/*.js', '**/*.jsx', '**/*.ts', '**/*.tsx') }}
+        restore-keys: |
+          ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') }}-
+    - name: Test
+      run: |
+        make web-build
+        go test -tags=${{ inputs.tags }} -timeout 15m -coverpkg=./... -coverprofile=${{ inputs.coverprofile }} -covermode=count ./...
+    - name: Get total code coverage
+      if: github.event_name == 'pull_request'
```

Comment:
Admittedly this wasn't something I thought much about, but eventually we'll want to calculate test coverage for the server/client binaries separately since they'll be looking at different packages. I'd rather keep this as is so that that's easier to do when we get there.

Reply:
Fair